ddl/partition.go

```diff
@@ -3146,7 +3162,7 @@ func checkPartitioningKeysConstraints(sctx sessionctx.Context, s *ast.CreateTabl
 			Length: types.UnspecifiedLength,
 		}}
 		if !checkUniqueKeyIncludePartKey(partCols, indexCols) {
-			return dbterror.ErrUniqueKeyNeedAllFieldsInPf.GenWithStackByArgs("PRIMARY KEY")
+			return dbterror.ErrUniqueKeyNeedAllFieldsInPf.GenWithStackByArgs("CLUSTERED KEY")
```

Comment:
Because we support global index for nonclustered primary key. I think `CLUSTERED KEY` may be more accurate than `primary key`

Reply:
Where are the restrictions of global index? (not sure I know where to look...)

What if the clustered index has multiple columns? Like:
```
create table tp (a int, b int, c varchar(244), primary key (a,b) clustered, key (c));
```
Will the global index implementation only work with a single column PK in case of a clustered index table?

Reply:
In the line above, `tblInfo.GetPkName()` seems like an odd function, only returning the first column of the primary key, so it can only be used if `PKIsHandle` is true? (i.e. "PKIsHandle is true when primary key is a single integer column.")

Then I still think that "PRIMARY KEY" is correct instead of "CLUSTERED INDEX", since all the partitioning columns are not a part of the primary key. I would be OK if it was "PRIMARY KEY, CLUSTERED INDEX" for more insights, but not really needed I think.

Why is not global index supported when PKIsHandle? (if it just is not implemented, that is OK, we can look into extending it later :) )

Reply:
I misunderstand with `PKIsHandle`.

Updated the PR, also return error when `tblInfo.IsCommonHandle = true`.